core/src/main/java/org/elasticsearch/index/query/NestedQueryBuilder.java

```diff
@@ -44,38 +44,20 @@
     public static final String NAME = "nested";
     public static final ParseField QUERY_NAME_FIELD = new ParseField(NAME);
 
-    /**
-     * The default score move for nested queries.
-     */
-    public static final ScoreMode DEFAULT_SCORE_MODE = ScoreMode.Avg;
-
     private static final ParseField SCORE_MODE_FIELD = new ParseField("score_mode");
     private static final ParseField PATH_FIELD = new ParseField("path");
     private static final ParseField QUERY_FIELD = new ParseField("query");
     private static final ParseField INNER_HITS_FIELD = new ParseField("inner_hits");
 
-    private final QueryBuilder<?> query;
-
     private final String path;
-
-    private ScoreMode scoreMode = DEFAULT_SCORE_MODE;
-
-    private InnerHitBuilder innerHitBuilder;
-
-    public NestedQueryBuilder(String path, QueryBuilder<?> query) {
-        if (path == null) {
-            throw new IllegalArgumentException("[" + NAME + "] requires 'path' field");
-        }
-        if (query == null) {
-            throw new IllegalArgumentException("[" + NAME + "] requires 'query' field");
-        }
-        this.path = path;
-        this.query = query;
-    }
+    private final ScoreMode scoreMode;
```

Comment:
QueryBuilders should always set the defaults since they can be constructed without ever passing through the parser (Java API)